### PR TITLE
[faucet] update API to use auth key instead of address

### PR DESCRIPTION
**Pull request by phoenix-antigravity**
creating new account by minting now requires to pass original auth key
so switching API to use that instead of address

### CODE REVIEW

File: client/cli/src/client_proxy.rs [hunk 1]
```diff
@@ -321,7 +321,7 @@ impl ClientProxy {
                 ),
                 is_blocking,
             ),
-            None => self.mint_coins_with_faucet_service(&receiver, num_coins, is_blocking),
```
Comment (sblackshear) [inline client/cli/src/client_proxy.rs]: Oops, I should have fixed this in the short addresses change. My bad

File: docker/mint/server.py [hunk 2]
```diff
@@ -45,7 +45,7 @@ def send_transaction():
     address = flask.request.args['address']
 
     # Return immediately if address is invalid
-    if re.match('^[a-f0-9]{32}$', address) is None:
+    if re.match('^[a-f0-9]{64}$', address) is None:
```
Comment (sblackshear) [inline docker/mint/server.py]: I would still expect this to be 32 bytes... (if that's what the 32 means)...

File: client/cli/src/client_proxy.rs [hunk 3]
```diff
@@ -1119,7 +1119,7 @@ impl ClientProxy {
             format!("http://{}", self.faucet_server).as_str(),
             &[
                 ("amount", num_coins.to_string().as_str()),
-                ("address", format!("{:?}", receiver).as_str()),
+                ("address", &hex::encode(receiver)),
```
Comment (sblackshear) [inline client/cli/src/client_proxy.rs]: Should this param name be changed to `autentication_key` (or TODO for changing it later)?

File: docker/mint/server.py [hunk 4]
```diff
@@ -45,7 +45,7 @@ def send_transaction():
     address = flask.request.args['address']
```
Comment (sblackshear) [inline docker/mint/server.py]: address -> authentication_key to  avoid confusion?

File: client/cli/src/client_proxy.rs (hunk 1 above)
Comment (phoenix-antigravity) [reply client/cli/src/client_proxy.rs]: no worries, we don't have proper test coverage for faucet service. It's easy to miss that. Speculation in a past was that it's a temporary thing. Sorry for bad experience

File: docker/mint/server.py (hunk 2 above)
Comment (phoenix-antigravity) [reply docker/mint/server.py]: hex encoded version of auth key has len of 64 from what I've seen

**Pull request closed without merging**